test/pytest/test_bridge.py

```diff
@@ -419,6 +419,42 @@ async def test_fsread1_errors(transport):
                                reply_keys={'message': "attribute 'max_read_size': must have type int"})
 
 
+@pytest.mark.asyncio
+async def test_fsread1_size_hint(transport):
+    transport.send_open('fsread1', path='/etc/passwd', binary='raw')
+    stat = os.stat('/etc/passwd')
+    while True:
+        channel, msg = await transport.next_frame()
+        if channel == '':
+            control = json.loads(msg)
+            command = control['command']
+            if command == 'ready':
+                assert control['size-hint'] == stat.st_size
+                break
+
+    # non-binary fsread1 has no size-hint
```

Comment:
You can't use `reply_keys` here because there's no way to assert the absence of a key.  Maybe you want to add something?  Or maybe you can just use `assert_msg('', command='ready', channel=channel)` and ensure that there's no `size-hint` in the returned dict.  Either way: I don't like that you leave the channel open and spewing data and then move on to the next case.